src/main/java/frc/robot/commands/Shoot.java

```diff
@@ -104,27 +154,34 @@ public void execute() {
                 //                         drive.getChassisSpeeds().vxMetersPerSecond+" "+
                 //                         drive.getChassisSpeeds().vyMetersPerSecond
                 // );
+
                 // TODO: Figure out what v_note is empirically
                 double v_note = 15;
 
-                // X distance to speaker
-                double x = Math.sqrt((displacement.getX() * displacement.getX())
-                                // Y distance to speaker
-                                + displacement.getY() * displacement.getY());
+                // X distance to speaker (along the floor to center of speaker)
+                double x = Math.hypot(displacement.getX(), displacement.getY());
                 // height (sorry that it's called y)
+                // TODO: but y is negative from above
```

Comment:
should y be -getDisplacementZ then?

Reply:
Not now. I'd like the formulas to be sensible, but changing things right now is not a good idea. I did not name y, I do not know where the equations came from; I've puzzled out a few. The negative value of y is compensated in some formulas that follow (IIRC, the formula for the z time). Negative values are also accounted for in some other places (e.g., calculating the horizontal angle also must compensate for negative getX() and negative getY() displacements).

The formulas are also loose with the exit height.

The formulas should also methods that can be tested without instantiating the subsystems. I want the unit tests in place so that transformation does not introduce even more errors.